test/image-blueprints/layer5-bootc/group2/cos9-bootc-source-flannel.containerfile

```diff
@@ -0,0 +1,20 @@
+FROM localhost/cos9-bootc-source:latest
+
+# Build arguments
+ARG USHIFT_RPM_REPO_NAME=microshift-local
+ARG USHIFT_RPM_REPO_PATH=/tmp/$USHIFT_RPM_REPO_NAME
+
+# Copy the MicroShift repository contents
+COPY ./rpm-repos/$USHIFT_RPM_REPO_NAME $USHIFT_RPM_REPO_PATH
+
+# Copy repository configuration
+COPY ./bootc-images/$USHIFT_RPM_REPO_NAME.repo ./bootc-images/microshift-centos9-nfv.repo ./bootc-images/microshift-rhocp-y.repo \
+    /etc/yum.repos.d/
+
+# Print repository configuration contents.
+# Install MicroShift optional packages and cleanup.
+RUN dnf repoinfo --enabled && \
+    dnf install -y "microshift-flannel-{{ .Env.SOURCE_VERSION }}" && \
```

Comment:
We're getting a build error
```
cos9-bootc-source-flannel: No match for argument: microshift-flannel-4.18.0_0.nightly_arm64_2024_10_06_235641_20241008093323_f1f5d710e33b83dc2a90e9912b92db92554728be-1.el9
cos9-bootc-source-flannel: Error: Unable to find a match: microshift-flannel-4.18.0_0.nightly_arm64_2024_10_06_235641_20241008093323_f1f5d710e33b83dc2a90e9912b92db92554728be-1.el9
cos9-bootc-source-flannel: Error: building at STEP "RUN dnf repoinfo --enabled &&     dnf install -y "microshift-flannel-4.18.0_0.nightly_arm64_2024_10_06_235641_20241008093323_f1f5d710e33b83dc2a90e9912b92db92554728be-1.el9" &&     rm -vf /etc/yum.repos.d/microshift-*.repo &&     rm -rvf $USHIFT_RPM_REPO_PATH &&     dnf clean all": while running runtime: exit status 1
```

Reply:
@praveenkumar , I think it's because we're losing WITH_FLANNEL=1 flag in the CI builds that are performed inside a container.
We are passing the flag to [rpm](https://github.com/openshift/microshift/blob/main/Makefile#L290) and [srpm](https://github.com/openshift/microshift/blob/main/Makefile#L300) targets, but not to [rpm-podman](https://github.com/openshift/microshift/blob/main/Makefile#L321) target.

Reply:
I tried small snip and that worked so I thought this will also work but looks like it is not passed :( should we add that environment variable to `common.sh` ?

```
$ cat /tmp/test.sh 
#!/bin/bash

BUILD_DRY_RUN=${BUILD_DRY_RUN:-false}
dry_run() {
    ${BUILD_DRY_RUN} && echo "echo"
}

run_bootc_image_build() {
    $(dry_run) env WITH_FLANNEL=1 bash -x /tmp/foo.sh
}

run_bootc_image_build


$ cat /tmp/foo.sh 
#!/bin/bash

echo ${WITH_FLANNEL}


$ /tmp/test.sh 
+ echo 1
1

$ BUILD_DRY_RUN=true /tmp/test.sh
env WITH_FLANNEL=1 bash -x /tmp/foo.sh
```

Reply:
> @praveenkumar , I think it's because we're losing WITH_FLANNEL=1 flag in the CI builds that are performed inside a container. We are passing the flag to [rpm](https://github.com/openshift/microshift/blob/main/Makefile#L290) and [srpm](https://github.com/openshift/microshift/blob/main/Makefile#L300) targets, but not to [rpm-podman](https://github.com/openshift/microshift/blob/main/Makefile#L321) target.

Ah, will add that and try again.